### PR TITLE
Consolidating api patterns

### PR DESCRIPTION
**Pull request by diehlbw**
# Overview
Moves _api.py contents into api/ and firms up the pattern of exposing functions for public usage.

## Description of changes
Obviates _version.py, move source of truth into setup.cfg then load it via importlib  
Fixes some formatting on docstrings  
Changed pandas applymap to map in reports, following deprecation warning replacement

## Author Checklist
- [ ] Linting passes; run early with [pre-commit hook](https://pre-commit.com/#install).
- [ ] Tests added for new code and issue being fixed.
- [ ] Added type annotations and full numpy-style docstrings for new methods.
- [ ] Draft your news fragment in new `changelog/ISSUE.TYPE.rst` files; see [changelog/README.md](https://github.com/epic-open-source/seismometer/blob/main/changelog/README.md).

### CODE REVIEW

**Review by firste — CHANGES_REQUESTED** [review]
Looks good outside of the one comment

File: src/seismometer/api/reports.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    data = data[[target, score_column] + cohort_columns]
+    data = FilterRule.isin(target, (0, 1)).filter(data)
+    positive_samples = data[target].sum()
+    if min(positive_samples, len(data) - positive_samples) < sg.censor_threshold:
+        return template.render_censored_plot_message(sg.censor_threshold)
+
+    try:
+        altair_plot = fairness_audit_altair(
+            data, cohort_columns, score_column, target, score_threshold, metric_list, fairness_threshold
+        )
```
[the remark this answers is not in the document]
Comment (diehlbw) [reply src/seismometer/api/reports.py]: Good catch. removed, and cleaned up imports

**Review by firste — APPROVED** [review]

File: docs/reference/seismometer.rst [hunk 2]
```diff
@@ -62,12 +58,13 @@ Custom Visualization Controls
    explore.ExplorationScoreComparisonByCohortWidget
    explore.ExplorationTargetComparisonByCohortWidget
 
-Example Notebook APIs
-~~~~~~~~~~~~~~~~~~~~~
+Notebook and Configuration Helper APIs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. currentmodule:: seismometer.__init__
+.. currentmodule:: seismometer.api
```
Comment (gbowlin) [inline docs/reference/seismometer.rst:64]: can this be removed?

File: docs/user_guide/index.rst [hunk 3]
```diff
@@ -490,10 +490,8 @@ Then, follow the pattern of normal startup but specify your function in the :py:
 
 .. code-block:: python
 
-   from seismometer.configuration import ConfigProvider
-   from seismometer.data.loader import loader_factory
+   import seismometer.api as sm
    from seismometer.seismogram import Seismogram
```
Comment (gbowlin) [inline docs/user_guide/index.rst:494]: should we update this line, it feels odd to import Seismogram from a different place than API.

File: src/seismometer/api/__init__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+from .reports import *
+from .templates import *
+from .utils import *
+
+__all__ = ["ConfigProvider", "loader_factory", "Seismogram", "init_logger"]
+__all__.extend(explore.__all__)
+__all__.extend(plots.__all__)
+__all__.extend(reports.__all__)
+__all__.extend(templates.__all__)
+__all__.extend(utils.__all__)
```
Comment (gbowlin) [inline src/seismometer/api/__init__.py:19]: How do you feel about:

```python
for module in [explore, plots, reports, templates, utils]:
    __all__.extend(module.__all__)
```

Comment (diehlbw) [reply src/seismometer/api/__init__.py:19]: FINC

File: src/seismometer/report/fairness.py [hunk 5]
```diff
@@ -210,7 +210,7 @@ def fairness_table(
         )
         cohort_ratios = cohort_data.div(cohort_data.loc[cohort_data[COUNT].idxmax()], axis=1)
 
-        cohort_icons = cohort_ratios.drop(COUNT, axis=1).applymap(
+        cohort_icons = cohort_ratios.drop(COUNT, axis=1).map(
```
Comment (gbowlin) [inline src/seismometer/report/fairness.py:213]: image

Comment (gbowlin) [reply src/seismometer/report/fairness.py:213]: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.map.html

mentions that map was added in Pandas 2.10

Comment (diehlbw) [reply src/seismometer/report/fairness.py]: rebased and dropped that commit

**Review by gbowlin — APPROVED** [review]

**Pull request merged**